packages/astro/src/runtime/server/index.ts

```diff
@@ -124,24 +124,46 @@ export async function renderComponent(result: SSRResult, displayName: string, Co
 
   // Call the renderers `check` hook to see if any claim this component.
   let renderer: Renderer | undefined;
-  for (const r of renderers) {
-    if (await r.ssr.check(Component, props, children)) {
-      renderer = r;
-      break;
+  if (renderers.length === 1) {
+      renderer = renderers[0];
+  } else if (metadata.hydrate !== 'only') {
```

Comment:
Does this still work if there’s only one renderer (above `if` check)? Seems like we need to just not set a renderer for `client:only` right?

Reply:
If there's only one renderer, we don't need to loop through them to find a match in either case. The first renderer will always apply, even for `client:only`.